Modules/pyexpat.c

```diff
@@ -1538,6 +1549,9 @@ xmlparse_dealloc(PyObject *op)
         XML_ParserFree(self->itself);
     }
     self->itself = NULL;
+    if (self->parent != NULL) {
+        Py_CLEAR(self->parent);
+    }
```

Comment:
Actually, we can just call `Py_CLEAR(self->parent);` here (no need for `self->parent != NULL`).